module-4/backend/week_two.md

```diff
@@ -5,17 +5,33 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What is Webpack and why is it useful?
+*webpack is a bundler for modules. It bundles JavaScript files to be used in a browser*
 2. When do you want to use event delegation?
+*You can bind anevent handler to a single parent element, and that handler will get executed whenever the event occurs on any of its child *
 3. What's one difference between ES5 and ES6?
+*ES6 has more functionality and is simplier*
 4. How are you using the MVC design pattern in your Quantified Self project?
+*server.js as controller*
+*lib/models as models*
+*and html, css files in front end* 
 5. How do you execute raw SQL in node?
+*using database.raw*
 6. What's a callback function and what are some reasons when we use/need callback functions?
+*It's a function that is passed as an argument to another function and invoked there. When we want to use this function later*
 7. What is CORS?
+*CORS provides a method to use resources across domains using HTTP*
 8. What are some steps to avoid CORS?
+*I include var cors = require('cors') in front and back end of my project for them to comunicate between each other*
 
 #### Review  
 
 9. Why do people say "HTTP is stateless"?
+*it means the connection between the browser and the server is lost once the transaction ends*
 10. What is a RESTful API?
+*its application program interface that uses http request to get, post and delete data*
```

Comment:
it also follows RESTful conventions - like the http verb matching up with the CRUD action matching with the path in the url